Python/ceval.c

```diff
@@ -5617,6 +5688,7 @@ MISS_WITH_OPARG_COUNTER(STORE_SUBSCR)
 
 error:
         call_shape.kwnames = NULL;
+        call_shape.init_pass_self = false;
```

Comment:
Note to reviewers: We don't set `frame->self = NULL` here because that means exceptions will destroy self. E.g. consider this:
```python
class A:
  def __init__(self):
    try:
      A.a # Kaboom!
    except AttributeError:
      pass

for _ in range(10):
  print(A())
```